### ISSUE TITLE
the onSuccess property does not work in the Edit and Create component

### ISSUE THREAD

**Issue by baselga** [marmelab/react-admin#5785]
When the onSuccess property is added to the Edit or Create component, the SaveButton component stops working.

**Environment**

* React-admin version: 3.11.2
* Last version that did not exhibit the issue (if applicable): 3.10.4
* React version: 17.0.1
* Browser:
* Stack trace (in case of a JS error):

**Comment by djhi** [comment]
Please follow the issue template

**Comment by Dragonqos** [comment]
I'm observing the same issue.
When using "onSuccess" prop in "Edit/Create", after Submit no data sending to dataProvider anymore, instead of this got empty response object inside onSuccess func, and obviously component stops working. 

Sandbox: https://codesandbox.io/s/laughing-bardeen-mwq4s?file=/src/posts/Edit.js

**Comment by baselga** [comment]
**What you were expecting:**
I hope you will call the onSusscess property after saving the form.

**What happened instead:**
When the onSuccess property is added to the Edit or Create component, the SaveButton component stops working.

**Related code:**

```
import React from 'react'
import { Edit, SimpleForm } from "react-admin"
import Form from './form'

export default (props) => (
  <Edit titleSource="firstName" {...props} onSuccess={()=>console.log("onSuccess")}>
    <SimpleForm>
      <Form {...props} />
    </SimpleForm>
  </Edit>
)

```

**Environment**

* React-admin version: 3.11.2
* Last version that did not exhibit the issue (if applicable):3.10.4
* React version:17.01
* Browser: All
* Stack trace (in case of a JS error):

**Comment by fzaninotto** [comment]
The `Edit` view is `undoable` by default. Undoable save is triggered when the success notification disappears. In your example, the onSuccess side effect doesn't include a call to `notify`, therefore the notification never shows up, and the call to `dataProvider.update()` is never triggered. 

This is a known gotcha of the optimistic / undoable mode (it relies on the trigger of a notification), but not a bug.

You can fix it in your code by adding a call to `notify` in your `onSuccess` callback, with the 4th parameter (undoable) true. 

```diff
import React from 'react'
-import { Edit, SimpleForm } from "react-admin"
+import { Edit, SimpleForm, useNotify } from "react-admin"
import Form from './form'

export default (props) => {
+ const notify = useNotify();
  const onSuccess = () => {
    console.log("onSuccess");
+   notify('success', 'info', null, true);
  }
  return (
    <Edit {...props} onSuccess={onSuccess}>
      <SimpleForm>
        <Form />
      </SimpleForm>
    </Edit>
  );
}
```

You  can also use `<Edit undoable={false}>` .

That being said, I agree that the documentation at https://marmelab.com/react-admin/CreateEdit.html#onsuccess isn't clear enough about this point. I'm turning this into a documentation issue.

**Comment by baselga** [comment]
Ok! Thank you for your support

**Comment by wmwart** [comment]
There is another problem. `({data})` always `undefined`

> The onSuccess function receives the response from the dataProvider call (dataProvider.create() or dataProvider.update()), which is the created/edited record (see the dataProvider documentation for details)

React-admin version: 3.11.3

**Comment by wmwart** [comment]
leave here or create a new issue?

**Comment by WiXSL** [comment]
I believe this doc enhancement has been made with the new `mutationMode` prop.
So this issue can be closed.

**Comment by fzaninotto** [comment]
Fixed by #5799
